packages/mobile/src/transactions/feed/TransactionFeed.tsx

```diff
@@ -0,0 +1,128 @@
+import SectionHead from '@celo/react-components/components/SectionHead'
+import { toChecksumAddress } from '@celo/utils/lib/address'
+import React, { useEffect, useMemo, useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { SectionList, View } from 'react-native'
+import { useDispatch } from 'react-redux'
+import config from 'src/geth/networkConfig'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import useSelector from 'src/redux/useSelector'
+import { tokensByAddressSelector } from 'src/tokens/selectors'
+import { updateTransactions } from 'src/transactions/actions'
+import { TRANSACTIONS_QUERY } from 'src/transactions/feed/query'
+import TransferFeedItem from 'src/transactions/feed/TransferFeedItem'
+import NoActivity from 'src/transactions/NoActivity'
+import { transactionsSelector } from 'src/transactions/reducer'
+import { FeedType } from 'src/transactions/TransactionFeed'
+import { TokenTransaction } from 'src/transactions/types'
+import { groupFeedItemsInSections } from 'src/transactions/utils'
+import Logger from 'src/utils/Logger'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+const TAG = 'transactions/TransactionFeed'
+// Query poll interval
+export const POLL_INTERVAL = 10000 // 10 secs
+
+interface TransactionFeed {
+  tokenTransactionsV2: {
+    __typename: 'TokenTransactionsV2'
+    transactions: TokenTransaction[]
+  }
+}
+
+function useQueryTransactionFeed() {
+  const address = useSelector(walletAddressSelector)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  const dispatch = useDispatch()
+
+  // Update the counter variable every |POLL_INTERVAL| so that a query is made to the backend.
+  const [counter, setCounter] = useState(0)
+  useEffect(() => {
+    const timer = setInterval(() => setCounter((n) => n + 1), POLL_INTERVAL)
+    return () => clearInterval(timer)
+  }, [])
+
+  // TODO: Extract this to a more generic function/hook so that it can be reused
+  const { loading, error, result } = useAsync(async () => {
+    const response = await fetch(`${config.blockchainApiUrl}/graphql`, {
```

Comment:
Preparing the move away from apollo client, right? 😄

Reply:
Yes the Apollo lib we're using is quite outdated, I feel you about this pain point.
I'm all for removing it from the code base too. 👍